src/renderers/dom/shared/DOMProperty.js

```diff
@@ -13,6 +13,21 @@
 
 var invariant = require('fbjs/lib/invariant');
 
+var RESERVED_PROPS = {
+  children: true,
+  dangerouslySetInnerHTML: true,
+  key: true,
```

Comment:
Why are `key` and `ref` important here? I'd expect them to never be in props, and therefore, never be arguments to the "set attribute" methods. They exist on the object you *pass* to an element, but shouldn't be in `element.props`.